frontend/src/app/app.component.ts

```diff
@@ -131,5 +91,28 @@ export class AppComponent implements OnInit {
         return hour < 12 ? 'AM' : 'PM';
       }
     });
-   }
+  }
+
+  getBrowserName() {
+    const  userAgent = navigator.userAgent;
+    let browserInfo = userAgent.match (/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [];
+    let parsedBrowserInfo;
+
+    if (/trident/i.test(browserInfo[1])) {
+      parsedBrowserInfo = /\brv[ :]+(\d+)/g.exec(userAgent) || [];
+      return { name: 'IE', version: (parsedBrowserInfo[1] || '') };
+    }
+    if (browserInfo[1] === 'Chrome') {
+      parsedBrowserInfo = userAgent.match (/\bOPR|Edge\/(\d+)/);
+      if (parsedBrowserInfo != null)   {
+        return { name: 'Opera', version: parsedBrowserInfo[1] };
+      }
+    }
+    browserInfo = browserInfo[2] ? [browserInfo[1], browserInfo[2]] : [navigator.appName, navigator.appVersion, '-?'];
```

Comment:
@Dmac26  What is going on here, does this work? Are you intentionally setting `browserInfo` equal to `browserInfo[2]` before your ternary or should it be a `===` to compare the values?

Reply:
@briandavidson - While looking at this to determine what I had in mind with that code, I remembered that the majority of this code snippet was borrowed from another developer on Shadat's team. I didn't quite understand everything that was going on with it, so I took a bit of a deeper dive and then decided to rewrite most of it to suite our needs. I believe that what is going on in my new method is far more apparent and too the point. Let me know what you think!